### DISCUSSION TITLE
Convergence with multiple processors

### DISCUSSION THREAD

**Discussion by erikgus94** [idaholab/moose#19412] [Q&A Modules: Navier-Stokes] [answered]
Hi,

This is a follow up question for [Boussinesq convergence problem](https://github.com/idaholab/moose/discussions/19208). 

I can get convergence when I'm adjusting the mesh that @GiudGiud helped me to generate, see input file [here](https://github.com/erikgus94/Boussinesq-question/blob/81120badb487d6096fdf988bc5439b6926158859/boussinesq72.i), but only when I run the code on one or two processors. When I try using more than 2 processors the residuals just wont go down for the first time step. I wonder if there is something I can do to be able to use more processors because currently the code takes about 8 hours to run and I have a cluster with 32 processors available that I can't use to it's full extent.

**Comment by GiudGiud** [comment, marked as answer]
Hello

Can you please paste your convergence (or non-convergence) history here?

The problem is likely the preconditioning is getting worse with an increasing number of processes. If you copied over some examples, you likely have LU preconditioner, which wont scale. What is your [Executioner] block?

There's more information on this page
https://mooseframework.inl.gov/source/executioners/Steady.html

Guillaume

**Reply by erikgus94** [reply]
You were right. I had the lu preconditioner and changing it to bjacobi solved the problem.

Thank you for the help!
